internal/controller/drplacementcontrol.go

```diff
@@ -974,8 +995,19 @@ func (d *DRPCInstance) quiesceAndRunFinalSync(homeCluster string) (bool, error)
 		addOrUpdateCondition(&d.instance.Status.Conditions, rmn.ConditionAvailable, d.instance.Generation,
 			d.getConditionStatusForTypeAvailable(), string(d.instance.Status.Phase), "Starting quiescing for relocation")
 
-		// clear current user PlacementRule's decision
-		d.setProgression(rmn.ProgressionClearingPlacement)
+		// We are going to clear the placement, this is when ACM will start
+		// deleting the workloads from the current cluster. In case of
+		// discovered apps, we have to let the user know that they need to
+		// clean up the apps from the current cluster. So set the progression
+		// to wait on user to clean up. For non-discovered apps, we can set the
+		// progression to clearing placement.
+		if d.instance.Spec.ProtectedNamespaces != nil &&
+			len(*d.instance.Spec.ProtectedNamespaces) > 0 {
+			d.setProgression(rmn.ProgressionWaitOnUserToCleanUp)
```

Comment:
Adding this line and removing line 993 will cause the `WaitOnUser` progression to persist throughout the entire final sync operation. Instead, we want the progression to switch to `RunningFinalSync` as soon as the VRG begins the final sync.
To achieve this, consider adding another boolean field to the VRG status, perhaps named `ReadyToRunFinalSync`. This field would only be set to `true` once the workload has been cleaned up, thereby allowing the progression to advance to `RunningFinalSync`.
Note that the VRG does not initiate the final sync until the PVC is no longer in use, so this condition can be leveraged for setting `ReadyToRunFinalSync`.

Reply:
That's the intent: the `finalsync` will not begin until the workload is deleted.
Using a condition is an option, but the `DataReady` condition is becoming overloaded with multiple purposes. It might make sense to bring the wait directly into PrepareForFinalSync. However, for now, taking the simplest path to completion may be the best approach.

Reply:
Have made an update such that in the case of discovered apps, we will show the waitonusertocleanup progression and in managed it will be finalsync one. We will fix it for discovered apps later.